build/test.ps1

```diff
@@ -9,21 +9,48 @@ $all_ok = $True
 Write-Host "Testing IQ#:"
 
 function Test-One {
-    Param($project)
+    <#
+    .SYNOPSIS
+        Runs dotnet test for the specified project, optionally
+        split into separate runs using the specified filters.
+    .DESCRIPTION
+        This function always runs all tests contained in the specified project.
+        
+        If no testClassFilters are specified, then all tests are run as part of
+        a single dotnet test invocation.
+
+        If testClassFilters is not empty, then dotnet test will be run multiple times:
+        - once per filter using `--filter ClassName~filter1`, `--filter ClassName~filter2`,
+          etc., where `filter1`, etc. represents each of the items in testClassFilters.
+        - once with `--filter (ClassName!~filter1 & ClassName!~filter2)`, where the
+          filter string runs all remaining tests and excludes all of the tests that were
+          previously run.
+    #>
+    Param([string] $project, [string[]] $testClassFilters = @())
 
     Write-Host "##[info]Testing $project"
-    dotnet test $project `
-        -c $Env:BUILD_CONFIGURATION `
-        -v $Env:BUILD_VERBOSITY `
-        --no-build `
-        --logger trx `
-        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
-        /property:InformationalVersion=$Env:SEMVER_VERSION `
-        /property:Version=$Env:ASSEMBLY_VERSION
+    if ($testClassFilters) {
+        $filterArgs = $testClassFilters | foreach { "ClassName~$_" }
+        $filterArgs += "(" + (($testClassFilters | foreach { "(ClassName!~$_)" }) -join " & ") + ")"
+    }
+    else {
+        $filterArgs = @("FullyQualifiedName!=_fake_")
+    }
+    $filterArgs | foreach {
+        dotnet test $project `
```

Comment:
Can you add a message here saying what project you're testing and with what filter? It will help troubleshooting build failures.

Reply:
That message is already here, a few lines below: 
```
Write-Host "##vso[task.logissue type=error;]Failed to test $project with filter $_"
```